Sources/CollectionViewManager.swift

```diff
@@ -217,111 +274,215 @@ extension CollectionViewManager {
+            
+            if indexes.count == cellItems.count {
+                zip(cellItems, indexes).forEach { cellItem, index in
+                    sectionItem.cellItems[index] = cellItem
+                    cellItem.indexPath = .init(row: index, section: sectionItem.index)
+                }
+                let indexPaths: [IndexPath] = indexes.map { index in
+                    return .init(row: index, section: section)
+                }
+                collectionView?.reloadItems(at: indexPaths)
+            }
+            else {
+                var removeIndexPaths: [IndexPath] = []
+                let firstIndex = indexes[0]
+                indexes.sorted().reversed().forEach { index in
+                    sectionItem.cellItems.remove(at: index)
+                    removeIndexPaths.append(.init(row: index, section: section))
+                }
+                
+                let insertIndexPaths: [IndexPath] = Array(firstIndex..<firstIndex + cellItems.count).map { index in
+                    return .init(row: index, section: sectionItem.index)
+                }
+                sectionItem.cellItems.insert(contentsOf: cellItems, at: firstIndex)
+                
+                self?.recalculateIndexPaths(in: sectionItem)
+                collectionView?.deleteItems(at: removeIndexPaths)
+                collectionView?.insertItems(at: insertIndexPaths)
+            }
         }, completion: completion)
     }
     
-    /// Removes cell items, that are contained inside specified section item, and then removes cells at the corresponding locations.
+    /// Removes cell items and then removes cells at the corresponding locations.
     ///
     /// - Parameters:
     ///   - cellItems: Cell items to remove
-    ///   - sectionItem: Section item that contains cell items to remove
     ///   - completion: A closure that either specifies any additional actions which should be performed after removing.
-    open func remove(_ cellItems: [CellItem], from sectionItem: SectionItem, completion: Completion? = nil) {
-        let sectionIndex = _sectionItems.index { $0 === sectionItem }!
-
-        let indexPaths: [IndexPath] = cellItems.flatMap { cellItem in
-            guard let row = sectionItem.cellItems.index(where: {$0 === cellItem}) else {
-                fatalError("Unable to remove cell item which is not contained in this section item.")
+    open func remove(_ cellItems: [CellItem], completion: Completion? = nil) {
+        perform(updates: { [weak self] collectionView in
+            let indexPaths = cellItems.map { cellItem in
+                return cellItem.indexPath
             }
-            return IndexPath(item: row, section: sectionIndex)
-        }
-        
-        collectionView.performBatchUpdates({
-            self.collectionView.deleteItems(at: indexPaths)
-            indexPaths.forEach { sectionItem.cellItems.remove(at: $0.row) }
+            indexPaths.sorted().reversed().forEach { indexPath in
```

Comment:
`sorted(by: >)`?

Reply:
The reason of this operations is to perform removing section items from the end of the array to preserve order of index paths and avoid crashes and unstable behaviour.
Does `sorted(by: >)` do the same?

Reply:
`sorted(by: >)` will do the same as `.sorted().reversed()` but shorter and faster :)